app/response_processor.py

```diff
@@ -46,6 +48,30 @@ def process(self, message, tx_id=None):
 
         return
 
+    def _check_namespace_error(self, response):
+        """Takes a response from rrm receipt endpoint and examines the xml
+        to identify whether a RetryableError or QuarantinableError should
+        be raised"""
+        namespaces = {'error': 'http://ns.ons.gov.uk/namespaces/resources/error'}
+        tree = etree.fromstring(response.content)
+        element = tree.find('error:message', namespaces).text
+        elements = element.split('-')
+
+        if elements[0] == '1009':
+            stat_unit_id = elements[-1].split('statistical_unit_id: ')[-1].split()[0]
+            collection_exercise_sid = elements[-1].split(
+                'collection_exercise_sid: ')[-1].split()[0]
+
+            self.logger.error("Receipt rejected by endpoint",
+                              msg="No records were found on the man_ce_sample_map table",
```

Comment:
Is there an actual error returned from RRM that we can display rather than hardcoding? Shouldn't really be reporting technical details about another service unless they were supplied to us (as they may be misleading because we're making an assumption)

Reply:
By this point we have already parsed the xml for the error, this is a very specific failure mode that was requested by RRM for logging purposes, from what I can remember.